### PR TITLE
Jenkins: specify required java version for plist & launcher

### PR DESCRIPTION
**Pull request by ryenus**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Java 8 is required since Jenkins 2.54 and Jenkins LTS 2.60.1, however Java 9 is not supported yet, see:
- https://jenkins.io/blog/2017/04/10/jenkins-has-upgraded-to-java-8/
- https://jenkins.io/doc/upgrade-guide/2.60/
- https://issues.jenkins-ci.org/browse/JENKINS-40689

In case one have both java 8 and java 9 installed, it's necessary to specify which java version to use for Jenkins.

Please note this requires an improved version of `bin.write_jar_script`, see:  https://github.com/Homebrew/brew/pull/3656.

### CODE REVIEW

**Comment by vinyldarkscratch** [comment]
The test bot seems to be failing due to it not having that improved version of `bin.write_jar_script`.  @DomT4 has a response for you on the pull request for Brew itself that you should look at?

**Comment by ryenus** [comment]
@vinyldarkscratch thank you and you're right, so I responded. Let's wait for @DomT4 to decide.

And I wish there could be a way to tell the bot to hold and wait until the other PR is merged first :D

**Comment by vinyldarkscratch** [comment]
@ryenus Actually, there's a nice little phrase you can say to trigger a build right on the spot:
"@ BrewTestBot test this please" (without the space between the @ and the username, of course)

That'll tell the test bot to start building right away.  :wink:

**Comment by ryenus** [comment]
@vinyldarkscratch not exactly what I want but surely that sounds nice!

So we can tell the bot to test again when the time is right, that's cool! I'm glad to know this. BTW, do we really need to be that polite to a bot? 😃 Sounds like it must be a kind bot.

**Comment by vinyldarkscratch** [comment]
Haha, it definitely is -- considering it's working so hard to test our formulas, and build the bottles we need, we've gotta give it some 💜 and kindness for sure!

**Comment by DomT4** [comment]
> Let's wait for @DomT4 to decide.

To be clear on this front, I'm powerless these days. My self-chosen "role" is more or less to drift in & out of the project providing advice/past experience where I can, but I have no formal power to yay/nay anything.

> "@ BrewTestBot test this please"

Doesn't work unless the bot knows you have that power, I'm afraid 😉.

**Comment by vinyldarkscratch** [comment]
Really?  @DomT4, I'm able to ask the test bot to test things...guess I have the power!  😈

**Comment by DomT4** [comment]
Unless things have changed behind-the-scenes, it used to need to know you had the "right" to call it, otherwise it'd just blank ya 🙈.

**Comment by ilovezfs** [comment]
@apjanke any thoughts here?

File: Formula/jenkins-lts.rb
Comment (apjanke) [inline Formula/jenkins-lts.rb]: Correct/more readable calling form:
```
bin.write_jar_script libexec/"jenkins.war", "jenkins-lts", :java_version => "1.8"
```

File: Formula/jenkins-lts.rb
Comment (apjanke) [inline Formula/jenkins-lts.rb]: bin.write_jar_script libexec/"jenkins-cli.jar", "jenkins-lts-cli", :java_version => "1.8"

File: Formula/jenkins.rb
Comment (apjanke) [inline Formula/jenkins.rb]: bin.write_jar_script libexec/"jenkins.war", "jenkins", :java_version => "1.8"

File: Formula/jenkins.rb
Comment (apjanke) [inline Formula/jenkins.rb]: bin.write_jar_script libexec/"jenkins-cli.jar", "jenkins-cli", :java_version => "1.8"

**Review by apjanke — CHANGES_REQUESTED** [review]
`write_jar_script` needs to use the named-parameter calling form.

Aside from that, this seems to me like the right thing to do.

**Review by ilovezfs — CHANGES_REQUESTED** [review]
@ryenus please address @apjanke's feedback when you have a chance!

**Comment by ryenus** [comment]
@apjanke @ilovezfs yeah, I agree that using named parameter is better here, thanks!

**Review by ilovezfs — DISMISSED** [review]
must wait for brew 1.5.5

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
@ryenus can you push an update, so that the pull request has 2 commits, with one commit per formula? thanks!

**Comment by ryenus** [comment]
@fxcoudert done!

@ilovezfs thanks so much for helping move this forward!

**Comment by fxcoudert** [comment]
Thanks @ryenus !

**Pull request closed without merging**
